pkg/controllers/controller.go

```diff
@@ -44,7 +45,7 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
-		return reconcile.Result{}, err
+		return reconcile.Result{Requeue: true}, err
```

Comment:
If it's not found, why requeue?

Reply:
If it's not found, we won't requeue here. If we fail to get it due to something like an APIServer connection, we do want to requeue, hence putting this outside of the if statement.

Reply:
Oh I see. Nice catch!!!

Reply:
That said, I think controller runtime automatically causes a requeue if you return an err

Reply:
Yes. That's true, but I want to put it here so it's clearer when we re-queue or not to the reader.